Added documentation on the Cake for Rider plugin

nils-a opened the pull request:
As with the blog post, this is mostly information from the readme.

Things that need to be changed, after the release and before the merge:
- [x] There's a link (placeholder) to the plugin in the marketplace, that needs to be set correctly.

nils-a commented:
@pascalberger @gep13 could you please have a look at it?

pascalberger commented on input/docs/integrations/editors/rider/index.cshtml:
```diff
@@ -0,0 +1,28 @@
+
+<h1>Installation & configuration</h1>
+
+See <a href="https://plugins.jetbrains.com/marketplace" target="_blank">Cake plugin for Rider</a> for instructions how to install and configure the plugin.
```
Does it require a min version of Rider? If so add a annotation like done [here](http://localhost:5080/docs/integrations/editors/visualstudio/#installation-configuration)

pascalberger reviewed (commented):
@nils-a Thanks for this 👍 

I added some comments. How does it look with IntelliSense support in Rider?

nils-a commented:
@pascalberger thanks for the review - I will implement the changes.

There is currently no support for IntelliSense in Cake for Rider - I will add a page, similar to https://cakebuild.net/docs/integrations/editors/visualstudio/intellisense

nils-a commented:
@pascalberger I have added the correct url to the plugin in the Marketplace and rebased. 

We could, technically, merge this in as soon as the plugin version in the marketplace has been approved. From that point on it would be possible to use the link provided in this commit to download the pre-release version manually from the Marketplace and "install from disc". 
@pascalberger , @gep13 what do you think?

pascalberger commented on input/docs/integrations/editors/rider/run-configurations.md:
```diff
@@ -0,0 +1,125 @@
+Order: 20
+Title: Run configurations
```
I would suggest to call it `Rider run configuration` which would improve search results

```suggestion
Title: Rider run configurations
```

pascalberger commented on input/docs/integrations/editors/rider/run-configurations.md:
```diff
@@ -0,0 +1,125 @@
+Description: Run configurations for Cake tasks
+---
+
+In order to use the run configurations you must have Cake installed on your machine or in your solution.
```
I would suggest to use the same heading structure here as we have used for [Visual Studio](http://cakebuild.net/docs/integrations/editors/visualstudio/task-runner):

- `Configuring run configurations`
  - `Install Cake`
  - `Supported runners`
  - `Settings`  (instead of `Settings regarding the run configurations`)
- `Using run configurations`
  - `Creating run configurations`
  - `Editing run configurations`
  - `Running builds` (instead of `Cake tasks tool window`)

nils-a commented:
@pascalberger I added your suggestions.

pascalberger reviewed (approved):
LGTM and ready to be merged once extension is available

gep13 reviewed (approved):
LGTM!

nils-a commented:
I'll send a ping when the current uploaded version ([0.1.0-alpha.1](https://plugins.jetbrains.com/plugin/15729-cake-rider/versions/alpha/106855)) was approved (and then probably do a rebase, too).

nils-a commented:
:shipit: 

![image](https://user-images.githubusercontent.com/349188/103568971-ccb77b80-4ec6-11eb-9a4f-1c11757fa7a9.png)

pascalberger commented:
@nils-a your changes have been merged, thanks for your contribution 👍
